### PR TITLE
feat(sumologicextension): use install tokens for registration

### PR DESCRIPTION
**Pull request by swiatekm-sumo**

### CODE REVIEW

File: docs/Migration.md [hunk 1]
```diff
@@ -2008,8 +1997,8 @@ The following table shows the equivalent [user.properties][user.properties] for
 | user.properties key                           | The OpenTelemetry Collector Key                            |
 |-----------------------------------------------|------------------------------------------------------------|
 | `wrapper.java.command=JRE Bin Location`       | N/A                                                        |
-| `accessid=accessId`                           | `extensions.sumologic.access_id`                           |
-| `accesskey=accessKey`                         | `extensions.sumologic.access_key`                          |
+| `accessid=accessId`                           | N/A, OTC only supports installation tokens                 |
+| `accesskey=accessKey`                         | N/A, OTC only supports installation tokens                 |
```
Comment (sumo-drosiek) [inline docs/Migration.md:2001]: We should mention `extensions.sumologic.install_token` here

File: pkg/extension/sumologicextension/README.md [hunk 2]
```diff
@@ -28,9 +28,7 @@ and can be used as an authenticator for the
 
 ## Configuration
 
-- `access_id`: (required) access ID for Sumo Logic service, see
-  [help][credentials_help] for more details
-- `access_key`: (required) access key for Sumo Logic service, see
```
Comment (pmalek-sumo) [inline pkg/extension/sumologicextension/README.md:33]: Can we perhaps make it crossed out for now and marked as deleted? Perhaps that would help users that use it as of now 🤔

File: pkg/extension/sumologicextension/README.md [hunk 3]
```diff
@@ -28,9 +28,9 @@ and can be used as an authenticator for the
...
+- ~~`access_id`: (required) access ID for Sumo Logic service~~, see `install_token` below
```
Comment (sumo-drosiek) [inline pkg/extension/sumologicextension/README.md:31]:
```suggestion
- ~`access_id`: (required) access ID for Sumo Logic service~, see `install_token` below
```

File: pkg/extension/sumologicextension/README.md [hunk 4]
```diff
@@ -28,9 +28,9 @@ and can be used as an authenticator for the
...
+- ~~`access_key`: (required) access key for Sumo Logic service~~, see `install_token` below
```
Comment (sumo-drosiek) [inline pkg/extension/sumologicextension/README.md:32]:
```suggestion
- ~`access_key`: (required) access key for Sumo Logic service~, see `install_token` below
```

**Review by sumo-drosiek — APPROVED** [review]

File: docs/Migration.md [hunk 5]
```diff
@@ -2008,8 +1997,8 @@ The following table shows the equivalent [user.properties][user.properties] for
 | user.properties key                           | The OpenTelemetry Collector Key                            |
 |-----------------------------------------------|------------------------------------------------------------|
 | `wrapper.java.command=JRE Bin Location`       | N/A                                                        |
-| `accessid=accessId`                           | `extensions.sumologic.access_id`                           |
-| `accesskey=accessKey`                         | `extensions.sumologic.access_key`                          |
+| `accessid=accessId`                           | N/A, use `extensions.sumologic.install_token`              |
+| `accesskey=accessKey`                         | N/A, use `extensions.sumologic.install_token`              |
```
Comment (pmalek-sumo) [inline docs/Migration.md:2001]: How about:

```suggestion
| ~~`accessid=accessId`~~                         | N/A, use `extensions.sumologic.install_token`              |
| ~~`accesskey=accessKey`~~                       | N/A, use `extensions.sumologic.install_token`              |
```

**Comment by pmalek-sumo** [comment]
Please mark this commit as breaking change as that's what it is 

```
feat(sumologicextension)!: use install tokens for registration
```

**Review by pmalek-sumo — APPROVED** [review]

File: pkg/extension/sumologicextension/README.md [hunk 6]
```diff
@@ -67,16 +67,15 @@ and can be used as an authenticator for the
...
 [fields_help]: https://help.sumologic.com/Manage/Fields
 
 ## Example Config
 
 ```yaml
 extensions:
   sumologic:
-    access_id: aaa
-    access_key: bbbbbbbbbbbbbbbbbbbbbb
+    install_token: U1VNT1RhSHd3eU9oTDhHSkxmZjdhRG5lbFYzODA3Q3ZodHRwczovL3ZtYW5kYWwtdm1hbmRhbC1tYWMtZXZlbnRzLnN1bW9sb2dpYy5uZXQ=
```
Comment (pmalek-sumo) [inline pkg/extension/sumologicextension/README.md:78]: Is that a live token? 🙃 Can this be made to look a bit more like a dummy placeholder?

File: pkg/extension/sumologicextension/extension.go [hunk 7]
```diff
@@ -90,8 +90,10 @@ var errGRPCNotSupported = fmt.Errorf("gRPC is not supported by sumologicextensio
 var _ configauth.ClientAuthenticator = (*SumologicExtension)(nil)
 
 func newSumologicExtension(conf *Config, logger *zap.Logger) (*SumologicExtension, error) {
-	if conf.Credentials.AccessID == "" || conf.Credentials.AccessKey == "" {
-		return nil, errors.New("access_key and/or access_id not provided")
+	if conf.Credentials.InstallToken == "" {
+		if conf.Credentials.AccessID == "" || conf.Credentials.AccessKey == "" {
+			return nil, errors.New("access credentials not provided: need either install_token or access_id and access_key")
+		}
```
Comment (pmalek-sumo) [inline pkg/extension/sumologicextension/extension.go:96]: Can we add a log here stating that user is using access key+id and they are deprecated? That would encourage people to migrate and give them a clear warning that they should expect this to go away.

File: pkg/extension/sumologicextension/README.md (hunk 6 above)
Comment (swiatekm-sumo) [reply pkg/extension/sumologicextension/README.md:78]: Of course it's not live, it's an example I stole from backend tests. I wanted to show how an actual token looks like. Can replace with a placeholder if you feel strongly about it.

File: docs/Configuration.md [hunk 8]
```diff
@@ -43,8 +43,7 @@ exporters:
 
 extensions:
   sumologic:
-    access_id: <my_access_id>
-    access_key: <my_access_key>
+    install_token: <token>
```
Comment (pmalek-sumo) [inline docs/Configuration.md:46]: These changes in docs will be problematic **until** we deploy installation token support on prods but 🤷

File: docs/Migration.md [hunk 9]
```diff
@@ -2008,8 +1997,8 @@ The following table shows the equivalent [user.properties][user.properties] for
 | user.properties key                           | The OpenTelemetry Collector Key                            |
 |-----------------------------------------------|------------------------------------------------------------|
 | `wrapper.java.command=JRE Bin Location`       | N/A                                                        |
-| `accessid=accessId`                           | `extensions.sumologic.access_id`                           |
-| `accesskey=accessKey`                         | `extensions.sumologic.access_key`                          |
+| ~~`accessid=accessId`~~                         | N/A, use `extensions.sumologic.install_token`              |
+| ~~`accesskey=accessKey`~~                       | N/A, use `extensions.sumologic.install_token`              |
```
Comment (pmalek-sumo) [inline docs/Migration.md:2001]:
```suggestion
| ~~`accessid=accessId`~~                        | N/A, use `extensions.sumologic.install_token`              |
| ~~`accesskey=accessKey`~~                      | N/A, use `extensions.sumologic.install_token`              |
```

File: pkg/extension/sumologicextension/README.md (hunk 6 above)
Comment (pmalek-sumo) [reply pkg/extension/sumologicextension/README.md:78]: I'd prefer a placeholder like in other docs

File: docs/Migration.md [hunk 10]
```diff
@@ -2008,8 +1997,8 @@ The following table shows the equivalent [user.properties][user.properties] for
 | user.properties key                           | The OpenTelemetry Collector Key                            |
 |-----------------------------------------------|------------------------------------------------------------|
 | `wrapper.java.command=JRE Bin Location`       | N/A                                                        |
-| `accessid=accessId`                           | `extensions.sumologic.access_id`                           |
-| `accesskey=accessKey`                         | `extensions.sumologic.access_key`                          |
+| ~~`accessid=accessId`~~                        | N/A, use `extensions.sumologic.install_token`              |
+| ~~`accesskey=accessKey`~~                      | N/A, use `extensions.sumologic.install_token`              |
```
Comment (pmalek-sumo) [inline docs/Migration.md:2001]: Heh, one more to make the table pretty :) 

```suggestion
| ~~`accessid=accessId`~~                       | N/A, use `extensions.sumologic.install_token`              |
| ~~`accesskey=accessKey`~~                     | N/A, use `extensions.sumologic.install_token`              |
```

**Review by pmalek-sumo — APPROVED** [review]

File: pkg/extension/sumologicextension/README.md (hunk 6 above)
Comment (swiatekm-sumo) [reply pkg/extension/sumologicextension/README.md]: Ok, changed.

**Pull request merged**